app/controllers/api/v1/motorcycles_controller.rb

```diff
@@ -35,30 +35,33 @@ def create
 
   # UPDATE MOTORCYCLE
   def update
-    # do your work
+    @motorcycle = Motorcycle.find(params[:id])
+    update_cycle = params.require(:motorcycle)
+      .permit(:cylinder, :description, :model, :acceleration, :title, :price, :duration, :discount, images: [])
+      .merge(user: @user)
+    if @motorcycle.update(update_cycle)
+      render json: { motorcycle: @motorcycle, message: 'Motorcycle updated successfully!' }
+    else
+      render json: { errors: @motorcycle.errors.full_messages, message: 'Motorcycle not updated!' }
+    end
   end
 
   # DELETE MOTORCYCLE
   def destroy
-    if @motorcycle == @user.motorcycles.find_by(id: params[:id])
-      if @motorcycle.reserve == false
-        @motorcycle.destroy
-        render json: { message: 'Motorcycle deleted successfully!' }, status: :ok
-      else
-        render json: { message: 'This motorcycle is on reserve my another user!' }, status: :unprocessable_entity
-      end
+    @motorcycle = Motorcycle.all
+    @motorcycle = Motorcycle.find(params[:id])
```

Comment:
- [ ]  Please first compare the user in the token with the creator of the motorcycle if they are equal delete the motorcycle. If they are not equal. Please send an error message. Remember Only the creator of the motorcycle can delete the item.